feat(gatsby-recipes): add Chakra UI recipe

shoxton opened the pull request:
<!--
  Have any questions? Check out the contributing docs at https://gatsby.dev/contribute, or
  ask in this Pull Request and a Gatsby maintainer will be happy to help :)
-->

<!--
  Is this a blog post? Check out the docs at https://www.gatsbyjs.com/contributing/blog-contributions/, and please mention if the blog post is pre-approved
  by someone from Gatsby.
-->

## Description
This PR adds [Chakra UI](https://github.com/chakra-ui/chakra-ui) to the recipes list

<!-- Write a brief description of the changes introduced by this PR -->

### Documentation
You can read more about this recipe documentation [here](https://github.com/shoxton/gatsby-recipe-chakra-ui)
<!--
  Where is this feature or API documented?

  - If docs exist:
    - Update any references, if relevant. This includes Guides and Gatsby Internals docs.
  - If no docs exist:
    - Create a stub for documentation including bullet points for how to use the feature, code snippets (including from happy path tests), etc.
  - Tag @gatsbyjs/documentation for review, pairing, polishing of the documentation
-->

KyleAMathews commented:
Hey! Thanks for adding this. Tested it out and found a couple of issues. One of which was a bug in recipes which I fixed in https://github.com/gatsbyjs/gatsby/pull/27790

The second is when I did get the recipe to run with the above fix, things didn't work out-of-the-box — looking at the Chakra docs, we also need to install 3 emotion dependencies — https://github.com/chakra-ui/chakra-ui#installing-chakra-ui

shoxton commented:
> Hey! Thanks for adding this. Tested it out and found a couple of issues. One of which was a bug in recipes which I fixed in #27790
> 
> The second is when I did get the recipe to run with the above fix, things didn't work out-of-the-box — looking at the Chakra docs, we also need to install 3 emotion dependencies — https://github.com/chakra-ui/chakra-ui#installing-chakra-ui

Thanks for checking my code. I think i fixed this on https://github.com/gatsbyjs/gatsby/pull/27721/commits/5c995243a8037933ebe723589a60371ef4f41927 so i changed the title to be more clear.

I replaced the shadow logic which was using the `<GatsbyShadowFile>` api by the `<File>` api to cover the bug you mentioned, just like [Theme UI recipe](https://raw.githubusercontent.com/gatsbyjs/gatsby/master/packages/gatsby-recipes/recipes/theme-ui.mdx) does. I also added the missing packages.

If theres anything that im missing out, please let me know.

KyleAMathews commented:
Oh can you revert to use ShadowFile? The bug will be fixed shortly.

KyleAMathews commented on packages/gatsby-recipes/recipes/chakra-ui.mdx:
```diff
@@ -0,0 +1,37 @@
+
+## Creates gatsby-plugin-chakra-ui `theme.js` file so you can customize your own global styles
+
+<File
```
The fix is in

```suggestion
<GatsbyShadowFile theme="gatsby-plugin-chakra-ui" path="src/theme.js" />
```

KyleAMathews reviewed (approved):
Thanks!

shoxton commented:
My pleasure! Is it possible to add this recipe to the official menu list as stated on https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-recipes/recipes#gatsby-recipes-mdx-source ?
Thanks!

KyleAMathews commented:
Oh of course. I overlooked that. Please PR it to
https://github.com/gatsbyjs/gatsby/blob/d0c94e98f013e53d946e0e75a3e4c1df20ae4c04/packages/gatsby-recipes/src/recipes-list.js

--
Kyle Mathews

Blog: *http://bricolage.io <http://bricolage.io>*
Twitter: http://twitter.com/kylemathews


On Wed, Nov 11, 2020 at 6:11 PM André <notifications@github.com> wrote:

> My pleasure! Is it possible to add this recipe to the official menu list
> as stated on
> https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-recipes/recipes#gatsby-recipes-mdx-source
> ?
> Thanks!
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/gatsbyjs/gatsby/pull/27721#issuecomment-725782196>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAARLB74ZSZXAPOJO6CT6A3SPM733ANCNFSM4TEOUNKQ>
> .
>

shoxton commented:
Done! Just created a PR #28013 to add this recipe to the list
